### ISSUE TITLE
Add support for reproducibility on reproducible-central 

### ISSUE THREAD

**Issue by algomaster99** [chains-project/maven-lockfile#747]
We should create a commit **and** push it to the default branch so that our releases have a corresponding commit. This is good because it will be easy to integrate the project with [reproducible central](https://github.com/jvm-repo-rebuild/reproducible-central) to guarantee reproducibility.

This release may not be fixed, but for later releases we must have a commit.

**Comment by monperrus** [comment]
yes, we want to be compatible with reproducible central.

> Commit corresponding to latest release X does not exist in the main branch
where is this convention documented? enforced?

**Comment by algomaster99** [comment]
This is from the comment https://github.com/INRIA/spoon/issues/5760#issuecomment-2053693661

>  what I really need is either no beta to Maven Central, or if you want to publich to Maven Central, publish a git tag as you do with releases

**Comment by yogyagamage** [comment]
> This release may not be fixed, but for later releases we must have a commit.

The release commit will be added to the main branch automatically, when we run the release pipeline successfully. 
In this release, the pipeline failed at merging because it took some time to pass the checks. So, I manually merged the branch. I have missed this point, that we should include the release commit. I squashed the commits, that is why we don't have the corresponding commit in the main branch. But in the pipeline we merge with fast forward, so the commit will be included.

On a side note, we should fix the pipeline to wait until checks are passed to merge.

**Comment by monperrus** [comment]
there is already some files in reproducible-central about maven-lockfile:  https://github.com/jvm-repo-rebuild/reproducible-central/blob/master/content/io/github/chains-project/maven-lockfile/README.md

However, maven-lockfile is not reproducible.

**Comment by kthfre** [comment]
I don't think there is a good solution for this at the moment as long as quarkus is a dependency. The ordering of the content in a couple of class files produced during the build is non-deterministic which of course affects the checksum of the jar file of the github action module. See: [1](https://github.com/quarkusio/quarkus/issues/676), [2](https://github.com/quarkusio/quarkus/issues/1020), [3](https://groups.google.com/g/quarkus-dev/c/6geWtb5HOdc), etc. They seem to be [looking to improve on this](https://github.com/quarkusio/quarkus/issues/7027).

**Comment by monperrus** [comment]
thanks for the analysis. why do we need quarkus? can we remove the dependency and the corresponding feature?

**Comment by kthfre** [comment]
It is a dependency of the Java package used for interaction with github actions, which essentially is the entire (relatively small) application. I'm not familiar enough with the Java packages available for this, and if there are alternatives, whether it is "worth" switching is not something I can determine.

**Comment by LogFlames** [comment]
Not fixed, only added badge to readme.

**Comment by monperrus** [comment]
Per https://github.com/jvm-repo-rebuild/reproducible-central/blob/master/content/io/github/chains-project/maven-lockfile/README.md it is now reproducible.

**Comment by algomaster99** [comment]
I don't think so it is still reproducible. [`maven-lockfile-github-action-5.4.1.jar`](https://github.com/jvm-repo-rebuild/reproducible-central/blob/master/content/io/github/chains-project/maven-lockfile/README.md) is the only artifact that is unreproducible. Reasons would most likely be the same as documented [here](https://github.com/chains-project/reproducible-central/issues/6#issuecomment-2515307140).

**Comment by LogFlames** [comment]
The github action should not require jbang. I think it would be easier to implement directly in action.yml. The GithubAction.java is a wrapper around the plugin which adds arguments and prints nicely formatted Github Actions output.

**Comment by LogFlames** [comment]
With v5.5.2 we are now finally fully reproducible (see https://github.com/jvm-repo-rebuild/reproducible-central/blob/master/content/io/github/chains-project/maven-lockfile/README.md).
